src/Features/CSharp/Portable/InlineHints/CSharpInlineTypeHintsService.cs

```diff
@@ -75,14 +76,40 @@ public CSharpInlineTypeHintsService()
                     if (parameter?.ContainingSymbol is IMethodSymbol { MethodKind: MethodKind.AnonymousFunction } &&
                         IsValidType(parameter?.Type))
                     {
-                        return (parameter.Type, new TextSpan(parameterNode.Identifier.SpanStart, 0));
+                        return new(parameter.Type, new TextSpan(parameterNode.Identifier.SpanStart, 0), trailingSpace: true);
+                    }
+                }
+            }
+
+            if (forTargetTypedNewTypes || displayAllOverride)
```

Comment:
forImplicitObjectCreation (and similarly revise anything else that should use this name).